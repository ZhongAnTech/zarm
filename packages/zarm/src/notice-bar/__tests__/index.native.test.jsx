import React from 'react';
import { Alert } from 'react-native';
import { render } from 'enzyme';
import toJson from 'enzyme-to-json';
import NoticeBar from '../index.native';

describe('NoticeBar', () => {
  it('renders correctly', () => {
    const wrapper = render(<NoticeBar>错误色</NoticeBar>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('theme', () => {
    const wrapper = render(<NoticeBar theme="danger">This is a NoticeBar</NoticeBar>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('scroll', () => {
    const wrapper = render(
      <NoticeBar scrollable>
        各位zarmer请注意，本组件使用了自动滚动功能，更多用法请参见使用文档。
      </NoticeBar>,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('onClick', () => {
    const wrapper = render(
      <NoticeBar onClick={() => Alert.alert('click this message!')}>NoticeBar</NoticeBar>,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('closable', () => {
    const wrapper = render(<NoticeBar closable>可关闭</NoticeBar>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
