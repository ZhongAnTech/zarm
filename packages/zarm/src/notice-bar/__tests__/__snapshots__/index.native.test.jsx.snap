// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoticeBar closable 1`] = `
<react-native-mock>
  <react-native-mock>
    <react-native-mock>
      <react-native-mock>
        可关闭
      </react-native-mock>
    </react-native-mock>
  </react-native-mock>
  <react-native-mock>
    <react-native-mock>
      <react-native-mock>
        <react-native-mock />
        <react-native-mock />
      </react-native-mock>
    </react-native-mock>
  </react-native-mock>
</react-native-mock>
`;

exports[`NoticeBar onClick 1`] = `
<react-native-mock>
  <react-native-mock>
    <react-native-mock>
      <react-native-mock>
        <react-native-mock>
          NoticeBar
        </react-native-mock>
      </react-native-mock>
    </react-native-mock>
  </react-native-mock>
</react-native-mock>
`;

exports[`NoticeBar renders correctly 1`] = `
<react-native-mock>
  <react-native-mock>
    <react-native-mock>
      <react-native-mock>
        错误色
      </react-native-mock>
    </react-native-mock>
  </react-native-mock>
</react-native-mock>
`;

exports[`NoticeBar scroll 1`] = `
<react-native-mock>
  <react-native-mock>
    <react-native-mock>
      <react-native-mock>
        各位zarmer请注意，本组件使用了自动滚动功能，更多用法请参见使用文档。
      </react-native-mock>
    </react-native-mock>
  </react-native-mock>
</react-native-mock>
`;

exports[`NoticeBar theme 1`] = `
<react-native-mock>
  <react-native-mock>
    <react-native-mock>
      <react-native-mock>
        This is a NoticeBar
      </react-native-mock>
    </react-native-mock>
  </react-native-mock>
</react-native-mock>
`;
